### PR TITLE
feat: add stats container and hook

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This pr add two elements, useTenantStats hook and StatsContainer component, the first one will get the data from the 
`/eox-nelp/api/stats/v1/tenant-stats/` endpoint an the second one will render multiple card stats base on the hook result

Issue https://edunext.atlassian.net/browse/FUTUREX-442

## Examples
`<StatsContainer showCourses showInstructors showLearners showVideos showProblems />`
image

`<StatsContainer showCourses showVideos  />`
image

## How to test
1. set the following in you mfe config api 
```
        "STATS_SETTINGS": {
            "API_BASE_URL": "http://domain/eox-nelp/api/stats/v1",
            "COURSE_IMAGE_URL": "https://something/image.svg",
            "INTRUCTOR_IMAGE_URL": "https://something/image.svg",
            "LEARNER_IMAGE_URL": "",
            "PROBLEM_IMAGE_URL": "https://something/image.svg",
        }
```
2. Import component from essentials import {  StatsContainer } from '@edunext/frontend-essentials';
3. set component `<StatsContainer showCourses showVideos  />`

### CODE REVIEW

File: src/hooks/useTenantStats/useTenantStats.test.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  test('Error retriving stats', async () => {
+    const get = jest.fn(() => Promise.reject(new Error('not found')));
+    // eslint-disable-next-line no-console
+    console.error = jest.fn();
+    getAuthenticatedHttpClient.mockReturnValue({ get });
+
+    const { result, waitForNextUpdate } = renderHook(() => useTenantStats());
+    await waitForNextUpdate().catch(() => {});
+
+    const state = result.current;
```
Comment (johanv26) [inline src/hooks/useTenantStats/useTenantStats.test.tsx:55]: Error is set to zero values. So there is no difference between an org that doesn't have data(new org) and an error in the response of the API?

Comment (johanv26) [reply src/hooks/useTenantStats/useTenantStats.test.tsx:55]: I mean I think this is not bad, but may be a warning or something that could help to debug this behavior knowing if the api is wrong or only the org is new. Or that is its real data.

**Comment by andrey-canon** [comment]
@johanv26 please review it

File: src/hooks/useTenantStats/useTenantStats.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      // On success the state will be updated based on the API response.
+      let videos = 0;
+      let problems = 0;
+      const learners = data?.learners ?? 0;
+      const instructors = data?.instructors ?? 0;
+      const courses = data?.courses.total_courses ?? 0;
+      const metrics = data?.courses.metrics ?? [];
+
+      metrics.forEach((metric: Metric) => {
+        videos += metric.components.video ?? 0;
```
Comment (johanv26) [inline src/hooks/useTenantStats/useTenantStats.tsx:105]: I would wait if this calculation should be the final way.
https://github.com/eduNEXT/eox-nelp/pull/55#discussion_r1245692787

File: src/hooks/useTenantStats/useTenantStats.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * Function that specifies how the state gets updated.
+ * @param  {State} state   Current state.
+ * @param  {Action} action Describes kind of action to be executed.
+ * @return {Object}        Updated state based on the action type.
+ */
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'SET_TENANT_STATS':
+      return action.state;
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply src/hooks/useTenantStats/useTenantStats.tsx:54]: You have a valid point but that doesn't make any difference in this implementation, first the state just can be updated through this method, so if some one try to change the state has to add a new action and create a new reference and finally in this implementation this just happens once, when the component is rendered

**Comment by andrey-canon** [comment]
> After the updates think is ok, but where are the translations in the ar.json for problem, video... `src/components/Stats/messages.ts`

of course I can add them, could you share that with me ?

File: src/components/Stats/messages.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+const { defineMessages } = require('@edx/frontend-platform/i18n');
+
+const messages = defineMessages({
+  title: {
+    id: 'essentials.statsContainer.title',
+    defaultMessage: 'Statistics',
```
Comment (johanv26) [inline src/components/Stats/messages.ts:6]:
```suggestion
    defaultMessage: 'Stadistics',
```

**Review by johanv26 — APPROVED** [review]

**Comment by johanv26** [comment]
Translations tested.
image

**Pull request merged**